soroban-react-dapp/contracts/greeting/src/instruction/auth.rs

```diff
@@ -0,0 +1,37 @@
+use crate::constants::*;
+use soroban_sdk::{Address, Env, Vec};
+
+// add editors
+pub fn add_editor(env: Env, new_editor: Address) {
+    let storage = env.storage().instance();
+    let admin: Address = storage.get(&ADMIN).unwrap();
+    admin.require_auth();
+
+    let mut editors: Vec<Address> = storage.get(&EDITORS).unwrap_or(Vec::new(&env));
+    if !editors.contains(&new_editor) {
+        editors.push_front(new_editor);
+        env.storage().instance().set(&EDITORS, &editors);
```

Comment:
Vec stored in instance storage could lead to DoS. Not big deal since it's just the instantiator who is adding admins